test/metabase/lib/equality_test.cljc

```diff
@@ -618,3 +620,46 @@
                (lib.equality/find-matching-column (lib/ref created-at) columns)))
         (is (= ca-expr
                (lib.equality/find-matching-column (lib/ref ca-expr)    columns)))))))
+
+(deftest ^:parallel query=-test
+  (let [query (lib/aggregate lib.tu/venues-query (lib/sum (meta/field-metadata :venues :id)))]
+    (are [doc a b] (lib.equality/query= a b)
+      "fresh uuids"
+      query (lib.util/fresh-uuids query)
+
+      "pMBQL and legacy"
+      query (lib.convert/->legacy-MBQL query)
+
+      "legacy and pMBQL"
+      (lib.convert/->legacy-MBQL query) query
+
+      "legacy and legacy"
+      (lib.convert/->legacy-MBQL query) (lib.convert/->legacy-MBQL query)
+
+      "nils"
+      nil nil)))
+
+(deftest ^:parallel query=-with-fields-test
+  (let [query-without-fields lib.tu/venues-query
+        query-with-fields (lib/with-fields query-without-fields
+                                           (mapv #(meta/field-metadata :venues %)
+                                                 (meta/fields :venues)))
+        field-ids (keep #(:id (meta/field-metadata :venues %)) (meta/fields :venues))]
```

Comment:
This list of all fields on `:venues` is nearly duplicated; let's give it a name and reuse it for legibility.